internal/files/linkedfiles_test.go

```diff
@@ -0,0 +1,168 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package files
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLinkUpdateChecksum(t *testing.T) {
+	wd, err := os.Getwd()
+	assert.NoError(t, err)
+	basePath := filepath.Join(wd, filepath.FromSlash("testdata/links"))
+	outdatedFile, err := NewLinkedFile(filepath.Join(basePath, "outdated.yml.link"))
+	t.Cleanup(func() {
+		_ = WriteFile(filepath.Join(outdatedFile.WorkDir, outdatedFile.LinkFilePath), []byte(outdatedFile.IncludedFilePath))
+	})
```

Comment:
Nit. It could be nice if we could use some temporary location, so we don't need to modify and revert changes in the working copy.